src/libs/metadata.c

```diff
@@ -289,7 +289,13 @@ static void _write_metadata(dt_lib_module_t *self)
     DT_DEBUG_CONTROL_SIGNAL_RAISE(darktable.signals,
                                   DT_SIGNAL_METADATA_CHANGED, DT_METADATA_SIGNAL_NEW_VALUE);
 
-    dt_image_synch_xmps(d->last_act_on);
+
+    for(const GList *imgs = d->last_act_on; imgs; imgs = g_list_next(imgs))
+    {
+      const dt_imgid_t imgid = GPOINTER_TO_INT(imgs->data);
+      dt_image_user_action(imgid);
```

Comment:
maybe better to call this at the actual point where the metadata is set in `common/metadata.c` routine `_metadata_execute`. Doing this here means that from Lua the status won't be set I suppose.

Reply:
I checked that,

1) a building issue due to locale setting for dt-cli (not understood yet)
2) My main concern, i think this would not be "safe" in the way - "only user action" ?